File: src/test/java/org/example/bankramenserver/domain/push/scheduler/MonthlyReportReadyPushSchedulerTest.java
package org.example.bankramenserver.domain.push.scheduler;

import org.example.bankramenserver.domain.push.service.SendMonthlyReportReadyPushService;
import org.junit.jupiter.api.Test;

import java.time.Clock;
import java.time.Instant;
import java.time.YearMonth;
import java.time.ZoneId;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

class MonthlyReportReadyPushSchedulerTest {

    @Test
    void sendMonthlyReportReadyPushTargetsPreviousMonth() {
        SendMonthlyReportReadyPushService service = mock(SendMonthlyReportReadyPushService.class);
        Clock clock = Clock.fixed(Instant.parse("2026-06-01T00:00:00Z"), ZoneId.of("Asia/Seoul"));
        MonthlyReportReadyPushScheduler scheduler = new MonthlyReportReadyPushScheduler(service, clock);

        scheduler.sendMonthlyReportReadyPush();

        verify(service).execute(YearMonth.of(2026, 5));
    }
}

File: src/test/kotlin/org/example/balogserver/domain/push/scheduler/MonthlyReportReadyPushSchedulerTest.kt
package org.example.balogserver.domain.push.scheduler

import org.example.balogserver.domain.push.service.SendMonthlyReportReadyPushService
import org.junit.jupiter.api.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.verify
import java.time.Clock
import java.time.Instant
import java.time.YearMonth
import java.time.ZoneId

class MonthlyReportReadyPushSchedulerTest {
    @Test fun sendMonthlyReportReadyPushTargetsPreviousMonth() {
        val service = mock(SendMonthlyReportReadyPushService::class.java)
        MonthlyReportReadyPushScheduler(service, Clock.fixed(Instant.parse("2026-06-01T00:00:00Z"), ZoneId.of("Asia/Seoul"))).sendMonthlyReportReadyPush()
        verify(service).execute(YearMonth.of(2026, 5))
    }
}
